RecoEcal/EgammaCoreTools/src/EcalClustersGraph.cc

```diff
@@ -0,0 +1,438 @@
+#include "RecoEcal/EgammaCoreTools/interface/EcalClustersGraph.h"
+#include <cmath>
+
+using namespace std;
+using namespace reco;
+using namespace reco::DeepSCInputs;
+
+typedef std::shared_ptr<CalibratedPFCluster> CalibratedClusterPtr;
```

Comment:
I did not go through the implementation, but I'm wondering: the `CalibratedPFCluster` is just an `edm::Ptr<reco::PFCluster>` (_i.e._ 2 `short`s, 1 long, and 1 atomic pointer); is it actually useful to wrap it in a ` std::shared_ptr` ?  Or would it be OK to use it by value ?